File: 206-reverse-linked-list/reverse-linked-list.ts
/**
 * Definition for singly-linked list.
 * class ListNode {
 *     val: number
 *     next: ListNode | null
 *     constructor(val?: number, next?: ListNode | null) {
 *         this.val = (val===undefined ? 0 : val)
 *         this.next = (next===undefined ? null : next)
 *     }
 * }
 */

function reverseList(head: ListNode | null): ListNode | null {
    if (head === null || head.next === null) return head;
    let prevNode = null;
    let cur = head;

    while (cur !== null) {
        let temp = cur;
        cur = cur.next;
        temp.next = prevNode;
        prevNode = temp;
    }
    return prevNode;

};

File: 206-reverse-linked-list/reverse-linked-list.java
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode() {}
 *     ListNode(int val) { this.val = val; }
 *     ListNode(int val, ListNode next) { this.val = val; this.next = next; }
 * }
 */
class Solution {
    public ListNode reverseList(ListNode head) {
        return reverse(head);
    }

    private ListNode reverse(ListNode head){
        ListNode prev = null;
        ListNode curNode = head;
        while(curNode != null){
            ListNode temp = new ListNode();
            temp = curNode.next;
            curNode.next = prev;
            prev = curNode;
            curNode = temp;
        }
        return prev;
    }
}